packages/terra-data-grid/src/DataGrid.jsx

```diff
@@ -252,31 +255,46 @@ const DataGrid = forwardRef((props, ref) => {
       columnId: displayedColumns[newColIndex].id,
     };
 
-    if (makeActiveElement) {
-      let focusedCell;
-      if (isSection(newRowIndex)) {
-        [focusedCell] = grid.current.rows[newRowIndex].cells;
+    if (!makeActiveElement) {
+      return;
+    }
 
-        if (!focusedCell.hasAttribute('tabindex')) {
-          focusedCell = grid.current.rows[newRowIndex].querySelector('button');
-        }
-      } else {
-        // Set focus on input field (checkbox) of row selection cells.
-        focusedCell = grid.current.rows[newRowIndex].cells[newColIndex];
-        if (isRowSelectionCell(newColIndex) && focusedCell.getElementsByTagName('input').length > 0) {
-          [focusedCell] = focusedCell.getElementsByTagName('input');
-        }
+    let focusedCell;
 
-        // Set focus to column header button, if it exists
-        const isHeaderRow = (newRowIndex === 0 || (hasColumnHeaderActions && newRowIndex === 1));
```

Comment:
What happened to the header row logic?

Reply:
Since header rows with action buttons also only have a single button, this overlaps with the current story to add focus to single buttons in any row. The isHeaderRow condition was removed, the rest of the logic was tweaked to accommodate single buttons/hyperlinks in any row.

Reply:
I tested the action button behavior and confirm that it works as expected after that change. As action button is a `<button>` element and the behavior for it is now a default behavior, it makes sense that the `isHeaderRow` condition is no longer needed.